svf-llvm/lib/LLVMModule.cpp

```diff
@@ -153,24 +153,25 @@ void LLVMModuleSet::build()
 void LLVMModuleSet::createSVFDataStructure()
 {
     getSVFType(IntegerType::getInt8Ty(getContext()));
-    Set<const Function*> candidateDecls;
-    Set<const Function*> candidateDefs;
+    // Functions need to be retrieved in the order of insertion
+    std::vector<const Function*> candidateDecls;
+    std::vector<const Function*> candidateDefs;
 
     for (Module& mod : modules)
     {
         std::vector<Function*> removedFuncList;
         /// Function
         for (Function& func : mod.functions())
         {
-            if (func.isDeclaration())
+            if (func.isDeclaration() && (FunDefToDeclsMap.find(&func) != FunDefToDeclsMap.end() || mod.getName().str() != Options::ExtAPIInput()))
```

Comment:
Fix this and put the later conditions inside the if branch. Add comments to explain this.